packages/base/src/utils/time-range.ts

```diff
@@ -1,18 +1,42 @@
+
 export class TimeRange {
     private start: bigint;
     private end: bigint;
     private offset: bigint | undefined;
 
-    /**
-     * Constructor.
-     * @param start Range start time
-     * @param end Range end time
-     * @param offset Time offset, if this is defined the start and end time should be relative to this value
-     */
-    constructor(start: bigint, end: bigint, offset?: bigint) {
-        this.start = start;
-        this.end = end;
-        this.offset = offset;
+    constructor();
+    constructor(timeRangeString: TimeRangeString);
+    constructor(start: bigint, end: bigint, offset?: bigint);
+    constructor(a?: TimeRangeString | bigint, b?: bigint, c?: bigint) {
```

Comment:
This does work.  Apparently this is the common pattern for "overloading" the constructor in TypeScript / JavaScript, because proper overloading is not allowed.

We can define a number of constructors in our class but we have to give one common implementation for all the constructors defined inside this constructor we can write our own logic in TypeScript.